keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,590 @@
+When a Pod is deleted, the PVC owned by the Pod is also deleted. When `RemoveOnScaledown` 
+policy is set and the Statefulset gets deleted the PVCs also will get deleted 
+(similar to `RemoveonStatefulSetDeletion` policy).
+
+Current scaleset controller implementation ensures that the manually deleted pods are restored 
+before the scale down logic is run. This combined with the fact that the owner references are set 
+only before the scale down will ensure that manual deletions do not automatically delete the PVCs 
+in question.
+
+During scale-up, if a PVC has an OwnerRef that does not match the Pod, it 
+potentially indicates that the PVC is referred by the deleted Pod and is in the process of 
+getting deleted. Controller will exit the current reconcile loop and attempt to reconcile in the 
+next iteration. This avoids a race with PVC deletion.
+
+When `PersistentVolumeClaimReclaimPolicy` is set to `RemoveOnStatefulSetDeletion` the owner reference in 
+PVC points to the StatefulSet. When a scale up or down occurs, the PVC would remain the same. 
+PVCs previously in use before scale down will be used again when the scale up occurs. The PVC deletion 
+should happen only after the Pod gets deleted. Since the Pod ownership has `blockOwnerDeletion` set to 
+`true` pods will get deleted before the StatefulSet is deleted. The `blockOwnerDeletion` for PVCs will 
+be set to `false` which ensures that PVC deletion happens only after the StatefulSet is deleted. This 
+chain of ownership ensures that Pod deletion occurs before the PVCs are deleted.
+
+
+`Retain` `PersistentVolumeClaimReclaimPolicy` will ensure the current behaviour - no PVC deletion is performed as part
+of StatefulSet controller.
+
+In alpha release we intend to keep the `PersistentVolumeClaimReclaimPolicy` immutable after creation. 
+Based on user feedback we will consider making this field mutable in future releases.
+
+## Cluster role change for statefulset controller
+Inorder to update the PVC ownerreference, the `buildControllerRoles` will be updated with 
+`patch` on PVC resource.
+
+### Test Plan
+
+1. Unit tests
+
+1. e2e tests
+    - RemoveOnScaleDown
+      1. Create 2 pod stateful set, scale to 1 pod, confirm PV deleted
+      1. Create 2 pod stateful set, add data to PVs, scale to 1 pod, scale back to 2, confirm PV empty
+      1. Create 2 pod stateful set, delete stateful set, confirm PVs deleted
+      1.Create 2 pod stateful set, add data to PVs, manually delete one pod, confirm pod comes back and PV has data (PV not deleted)
+      1. As above, but manually delete all pods in stateful set
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, confirm PV is deleted
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, scale back to two pods, confirm PV is empty
```

Comment:
Suggest also testing that PVCs won't be deleted during a rolling updating for `RemoveOnScaleDown`.